src/api/app/models/status_history.rb

```diff
@@ -1,10 +1,17 @@
 class StatusHistory < ApplicationRecord
-  def self.history_by_key_and_hours(key, hours = 24)
-    starttime = Time.now.to_i - hours.to_i * 3600
-
-    where("time >= ? AND \`key\` = ?", starttime, key).
-      pluck(:time, :value).
-      collect { |time, value| [time.to_i, value.to_f] }
+  def self.history_by_key_and_hours(key, hours = 24, end_date = '')
+    if end_date.empty?
+      start_time = Time.now.to_i - hours.to_i * 3600
+      where("time >= ? AND \`key\` = ?", start_time, key).
+        pluck(:time, :value).
+        collect { |time, value| [time.to_i, value.to_f] }
+    else
```

Comment:
there are some code repeated in the if-else, why don't you put only the where and the `end_time` assignment in the if-else and the rest outside? this way you only write it once. :wink: